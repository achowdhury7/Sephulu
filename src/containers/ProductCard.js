import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import classNames from 'classnames/bind'

export const ProductCard = (props) => {
  const product = props.products[props.productId]
  const productPriceClasses = classNames({
    'product-price': true,
    'cancelled': product.under_sale
  })
  const productSalePriceClasses = classNames({
    'product-sale-price': true,
    hidden: !product.under_sale
  })

  return (
    <div className="product-container">
      <div className="product-image"></div> 
      <div className="product-desc">
        <div className="product-name"><span>{product.name}</span></div>
        <div className="product-category"><span>{product.category}</span></div>
        <div className="product-price-container">
          <span className={productSalePriceClasses}>${(product.sale_price/100).toFixed(2)}</span>
          <span className={productPriceClasses}>${(product.price/100).toFixed(2)}</span>
        </div>
      </div> 
    </div>
  )
}

const mapStateToProps = state => ({
  products: state.app.products.byId
})


export default connect(
  mapStateToProps
)(ProductCard)