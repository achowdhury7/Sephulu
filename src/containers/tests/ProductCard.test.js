import React from 'react'
import { shallow, mount, render } from 'enzyme'

import { ProductCard } from '../ProductCard'

let wrapper

beforeEach(() => {
  const mockProducts = {
    1: { id: '1', name: 'product1', category: 'product', under_sale: true } 
  }
  const mockProductId = '1'
  wrapper = shallow(
    <ProductCard products={mockProducts} productId={mockProductId} />
  )
})

test('render product card', () => {
  expect(wrapper).toMatchSnapshot()
})

test('when product.under_sale is true', () => {
  expect(wrapper.find('span.product-price').is('.cancelled')).toEqual(true)
})