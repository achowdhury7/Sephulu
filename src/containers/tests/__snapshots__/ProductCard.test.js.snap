// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render product card 1`] = `
<div
  className="product-container"
>
  <div
    className="product-image"
  />
  <div
    className="product-desc"
  >
    <div
      className="product-name"
    >
      <span>
        product1
      </span>
    </div>
    <div
      className="product-category"
    >
      <span>
        product
      </span>
    </div>
    <div
      className="product-price-container"
    >
      <span
        className="product-sale-price"
      >
        $
        NaN
      </span>
      <span
        className="product-price cancelled"
      >
        $
        NaN
      </span>
    </div>
  </div>
</div>
`;
